src/rpcmessage.rs

```diff
@@ -84,15 +116,19 @@ impl RpcMessage {
         self.set_ival(Key::Params, rv)
     }
     pub fn result(&self) -> Result<&RpcValue, RpcError> {
-        match self.ival(Key::Error as i32) {
-            None => {
-                Ok(self.ival(Key::Result as i32).unwrap_or(STATIC_NULL_RPCVALUE.get_or_init(RpcValue::null)))
-            }
-            Some(err) => {
-                Err(RpcError::from_rpcvalue(err)
-                    .unwrap_or(RpcError{ code: RpcErrorCode::InvalidParam, message: "Cannot parse 'error' key found in RPC response.".to_string() }))
-            }
+        if let Some(err) = self.ival(Key::Error as i32) {
+            return Err(RpcError::from_rpcvalue(err)
+                .unwrap_or(RpcError {
+                    code: RpcErrorCode::InvalidParam,
+                    message: "Cannot parse 'error' key found in RPC response.".to_string(),
+                }));
         }
+
+        self.ival(Key::Result as i32)
+            .ok_or_else(|| RpcError {
+                code: RpcErrorCode::InvalidParam,
+                message: "Missing 'result' key in RPC response.".to_string(),
+            })
```

Comment:
@fvacek this method now can't return `Result<&RpcValue, RpcError>`, because we have new a Response type - delay. How should we proceed? Also, `is_success` is now also wrong, because of this.

Reply:
I thing, we should return enum rpcvalue, error, delay

Reply:
Another issue is that introducing Delay messages alters the way responses are handled.

In the SHV specification, delays are intended to carry progress information, which clearly implies they should be propagated up the application stack to the callers. However, if we follow this approach, all callers would need to handle each response as either a delay or a final result. Consequently, the return type of an RPC call would shift from a single-response future to a stream. This would significantly complicate the API, especially considering that the majority of use cases involve simple calls that return a single result.

Reply:
Will be resolved in `libshvclient-rs` using an optional callback for progress tracking while keeping the almost unchanged API.